src/Message/PurchaseRequest.php

```diff
@@ -62,13 +66,51 @@ protected function getFunction()
         return 'authCapture';
     }
 
+    /**
+     * @param string $paymentMethodType
+     * @return array
+     */
     public function getData($paymentMethodType = self::PAYMENT_METHOD_CREDIT_CARD)
     {
         $data = parent::getData($paymentMethodType);
 
-        $data['ignoreAvsPolicy'] = false;
-        $data['ignoreCvnPolicy'] = false;
+        $ignore_avs = $this->getIgnoreAvsPolicy();
+        $ignore_cvn = $this->getIgnoreCvnPolicy();
+        $data['ignoreAvsPolicy'] = $ignore_avs ? $ignore_avs : false;
+        $data['ignoreCvnPolicy'] = $ignore_cvn ? $ignore_cvn : false;
 
         return $data;
     }
+
+    /**
+     * @return null|bool
+     */
+    public function getIgnoreAvsPolicy()
+    {
+        return $this->getParameter('ignoreAvsPolicy');
+    }
+
+    /**
+     * @param bool $ignore
```

Comment:
Psalm will complain about not having an `@return` here actually

Reply:
I think you have to do:
```
@return static
```